extensions/slack/src/search-messages.tsx

```diff
@@ -0,0 +1,164 @@
+      const webClient = getSlackWebClient();
+
+      const results = await webClient.search.messages({
+        query: `${query}${channel ? ` in:${channel}` : ""}`,
+        sort,
+      });
+
+      return results.messages?.matches ?? [];
+    },
+    [query, selectedChannel, sortType],
+    {
+      onError(error) {
+        handleError(error);
+      },
+      execute: query.length > 0,
+      keepPreviousData: true,
+    },
+  );
+
+  const sortOptions: { title: string; sort: SearchMessagesArguments["sort"] }[] = [
+    { title: "Newest", sort: "timestamp" },
+    { title: "Most Relevant", sort: "score" },
+  ];
+
+  const FromActions = () => (
+    <>
+      <Action
+        title="From Me Only"
+        icon={meInfo ? meInfo.icon : Icon.Person}
+        shortcut={{ modifiers: ["cmd", "shift"], key: "i" }}
+        onAction={() => setQuery(query ? `from:me ${query}` : "from:me ")}
+      />
+
+      <ActionPanel.Submenu icon={Icon.Person} title="From" shortcut={{ modifiers: ["cmd", "shift"], key: "u" }}>
+        {users?.map((u) => (
+          <Action
+            key={u.id}
+            title={u.name}
+            icon={u.icon}
+            onAction={() => setQuery(query ? `from:${u.name} ${query}` : `from:${u.name} `)}
+          />
+        ))}
+      </ActionPanel.Submenu>
+    </>
+  );
+
+  return (
+    <List
+      isLoading={isLoading}
+      searchText={query}
+      onSearchTextChange={setQuery}
+      actions={
+        <ActionPanel>
+          <FromActions />
+        </ActionPanel>
+      }
+      throttle
+      isShowingDetail={data && data.length > 0}
+      searchBarAccessory={
+        <List.Dropdown tooltip="Search Channels" onChange={setSelectedChannel}>
+          <List.Dropdown.Item value="" title="All Channels" />
+          {channels ? (
+            <List.Dropdown.Section>
+              {channels.flat().map((c) => {
+                return <List.Dropdown.Item key={c.id} icon={c.icon} value={c.name} title={c.name} />;
+              })}
+            </List.Dropdown.Section>
+          ) : null}
+        </List.Dropdown>
+      }
+    >
+      {data?.map((m) => {
+        if (!m.text || !m.ts) return null;
+
+        const user = users?.find((u) => u.id === m.user);
+        const date = convertTimestampToDate(m.ts);
+        const text = emoji.emojify(m.text);
+        const formattedDate = format(date, "EEEE dd MMMM yyyy 'at' HH:mm");
+
+        return (
+          <List.Item
+            key={m.iid}
+            icon={{ value: user?.icon, tooltip: user?.name ?? "Unknown user" }}
+            title={text}
+            accessories={[{ date, tooltip: formattedDate }]}
+            detail={
+              <List.Item.Detail
+                markdown={text}
+                metadata={
+                  <List.Item.Detail.Metadata>
+                    {m.channel ? (
+                      <List.Item.Detail.Metadata.Label title="Channel Name" icon={Icon.Hashtag} text={m.channel.name} />
```

Comment:
Suggestion (non-blocking): If possible, we would want to render the channel differently for users, actual channels and MPIM(s). `channel.name` works well with channels but not with users and MPIM groups.